freqtrade/plugins/pairlist/MarketCapPairList.py

```diff
@@ -0,0 +1,92 @@
+    def __init__(self, exchange, pairlistmanager,
+                 config: Dict[str, Any], pairlistconfig: Dict[str, Any],
+                 pairlist_pos: int) -> None:
+        super().__init__(exchange, pairlistmanager, config, pairlistconfig, pairlist_pos)
+
+        if 'number_assets' not in self._pairlistconfig:
+            raise OperationalException(
+                '`number_assets` not specified. Please check your configuration '
+                'for "pairlist.config.number_assets"')
+
+        self._stake_currency = config['stake_currency']
+        self._number_pairs = self._pairlistconfig['number_assets']
+        self._refresh_period = self._pairlistconfig.get('refresh_period', 6*60*60)
+        self._pair_cache: TTLCache = TTLCache(maxsize=1, ttl=self._refresh_period)
+        self._url = \
+            f'https://api.coingecko.com/api/v3/coins/markets?vs_currency=usd&order=market_cap_desc&per_page={self._number_pairs}&page=1&sparkline=false'   # noqa
+        if not self._exchange.exchange_has('fetchTickers'):
+            raise OperationalException(
+                'Exchange does not support dynamic whitelist. '
+                'Please edit your config and restart the bot.'
+            )
+
+    @property
+    def needstickers(self) -> bool:
+        """
+        Boolean property defining if tickers are necessary.
+        If no Pairlist requires tickers, an empty Dict is passed
+        as tickers argument to filter_pairlist
+        """
+        return True
+
+    def short_desc(self) -> str:
+        """
+        Short whitelist method description - used for startup-messages
+        """
+        return f"{self.name} - top {self._pairlistconfig['number_assets']} marketcap pairs."
+
+    def gen_pairlist(self, tickers: Dict) -> List[str]:
+        """
+        Generate the pairlist
+        :param tickers: Tickers (from exchange.get_tickers()). May be cached.
+        :return: List of pairs
+        """
+        # Generate dynamic whitelist
+        # Must always run if this pairlist is not the first in the list.
+        pairlist = self._pair_cache.get('pairlist')
+        if not pairlist:
+            resp = requests.get(self._url)
+            data = resp.json()
+            pairlist = [entry['symbol'].upper() + f'/{self._stake_currency}' for entry in data]
+            pairlist = self.filter_pairlist(pairlist, tickers)
+            self._pair_cache['pairlist'] = pairlist
+
+        return pairlist
+
+    def filter_pairlist(self, pairlist: List[str], tickers: Dict) -> List[str]:
+        """
+        Filters and sorts pairlist and returns the whitelist again.
+        Called on each bot iteration - please use internal caching if necessary
+        :param pairlist: pairlist to filter or sort
+        :param tickers: Tickers (from exchange.get_tickers()). May be cached.
+        :return: new whitelist
+        """
+        # Use the incoming pairlist.
+        pairlist = [k for k in pairlist if k in tickers.keys()]
```

Comment:
You're running under the assumption that this pairlist is the FIRST pairlist in your list - which is not true.


Assuming you use VolumePairlist (get the highest volume pairs) - followed by MarketCapPairlist to sort the top 30 volumePairs based on their market cap.

This means - the "incomming" pairlist (`filter_pairlist`) will not be filterd based on MarketCap - but by "something" else (in the above sample by volume - but other options are possible easily).

As such - this `filter_pairlist` will not do anything - as it does completely ignore the MarketCap of the pairs - so the sequence of the pairs will not be changed.

Reply:
Indeed failed to understand how the pairlists interact. Fixed.